### ISSUE TITLE
Instancing an Area2D from body_entered throws an error (regression from 0b1e93c)

### ISSUE THREAD

**Issue by nazgum** [godotengine/godot#26148]
**Godot version:**
3.1 nightly (hugo from feb 15)

**OS/device including version:**
Ubuntu 18.10

**Issue description:**
On the body_entered signal for an Area2D, if you try to instance another scene that contains an Area2D, it errors out.

```
extends Area2D

const spawned_area_scene = preload("SpawnedArea.tscn")

func _ready():
	connect("body_entered", self, "_on_body_enter")

func _on_body_enter(target):
	print("entered")
	var spawn_area = spawned_area_scene.instance()
```

This is the error that gets shown.

```
E 0:00:01:0920   Function blocked during in/out signal. Use set_deferred("monitorable",true/false)
  <C Source>     scene/2d/area_2d.cpp:432 @ set_monitorable()
  <Stack Trace>  Area1.gd:11 @ _on_body_enter()
```

**Minimal reproduction project:**
[broken_area2d.zip](https://github.com/godotengine/godot/files/2892153/broken_area2d.zip)

**Comment by akien-mga** [comment]
That's due to 0b1e93ccd43815d93de031b4f23fbfcd9ae7a34c. It should probably show this error only when you change the `monitorable` property of the node receiving the signal, not of another node (here, a new one getting instanced, and having its `monitorable` property initialized).

**Comment by nazgum** [comment]
Just tested this again with hugo nightly from today (feb 23), and this is still broken but now throws a different error (with the same code/sample project from above):

```
E 0:00:04:0523   Condition ' locked || Physics2DServer::get_singleton()->is_flushing_queries() ' is true.
  <C Source>     scene/2d/area_2d.cpp:432 @ set_monitorable()
  <Stack Trace>  ArrowTarget.gd:16 @ _on_body_enter()

E 0:00:04:0523   Can't change this state while flushing queries. Use call_deferred() or set_deferred() to change monitoring state instead
  <C Source>     servers/physics_2d/physics_2d_server_sw.cpp:413 @ area_set_shape_disabled()
  <Stack Trace>  ArrowTarget.gd:16 @ _on_body_enter()
```

**Comment by reduz** [comment]
Fixed by b84b015
